src/app/icd/ICDMonitoringTable.h

```diff
@@ -59,6 +59,25 @@ struct ICDMonitoringEntry : public PersistentData<kICDMonitoringBufferSize>
     CHIP_ERROR Serialize(TLV::TLVWriter & writer) const override;
     CHIP_ERROR Deserialize(TLV::TLVReader & reader) override;
     void Clear() override;
+    /**
+     * @brief Set the Key object
+     *        This method will create a new keyHandle. The key handle might contain either
+     *        the raw key or a keyID depending on which Crypto implementation is used.
+     *        In any case, to prevent key leakage, one should either call the DeleteKey method
+     *        or save the entry within the ICDMonitoring Table before this object goes out of scope.
+     *
+     *        In some implementation Calling SetKey() twice on the same ICDMonitoringEntry will result
+     *        in the first Key being deleted from the keyStore to prevent key leakage and this even if
+     *        the entry was previously saved in the table since the key storage is decoupled from the
+     *        ICDMonitoringTableStorage. One should use a new object or manually clear the content of
+     *        the key handle prior to calling SetKey() again.
```

Comment:
```suggestion
     *       Calling SetKey() on an ICDMonitoringEntry that already has a valid key handle will lead to
     *       the old key being deleted if the key handle is a key id, not raw data.  This can lead to keys
     *       that were already persisted via the ICDMonitoring table getting lost.
     *
     *       Therefore a new entry object should be used for each key when adding entries to the ICDMonitoring
     *       table.
```

Because again, it's not cler how to "manually clear" things here.

But I do wonder: should we just fail out of SetKey happens when we have a key already?  That is, should we have a SetKey that only lets you set a key if there isn't one, and a ReplaceKey that lets you replace (and makes it clear it's replacing)?

Reply:
That was my initial approach until I realized that there is no way of knowing if a key handle is empty or not... 

The Key handle buffer could have some data in it, it could have a Zero key, it could have a KeyID 0.... That would require reworking part of the CryptoPAL API to add either a bool to the key handle or something else to expose if the buffer within the KeyHandle is populated or not...

Reply:
Added a isKeySet flag